### ISSUE TITLE
TypeScript version

### ISSUE THREAD

**Issue by dsheyp** [ungap/event-target#7]
Hi

In the process of "typing" your event-target polyfill I came across the following issues:

The attempt to delete these fields produces errors as they are read-only. So are they in plain JS. Does this mean these lines could be safely removed from the code?

```
        delete event.currentTarget;
        delete event.target;
```

The last line of the addEventListener definition produces an error, as `listeners` is only declared locally in the for-loop. Into which  context could the declaration for `listeners` be moved without breaking the code?

`
listeners.push({target: this, listener: listener, options: options});
`

Thanks for any suggestions
dsheyp

**Comment by dsheyp** [comment]
Hi

I ended up removing the before mentionend lines and I moved the declaration for `listeners` one level up, in front of the for-loop. Everything seems to work so far.

I guess my question about the declaration of `listeners` was really more for my own understanding of your code. You declare `listeners` in all three methods. But where to you read from `listeners`? This is the part I don't understand....

**Comment by WebReflection** [comment]
I don't think I understand any of your comments/questions ... listeners are always taken from a WeakMap, hence private per instance, and each different event type gets its own stack to add/remove callbacks or objects ... the code is tested/code-covered 100% and it has zero issues whatsoever, so I am not sure how to help here, also because I don't use TS at all, it's usually people using my projects that file PRs adding types when/if needed.

**Comment by dsheyp** [comment]
Thank you for the response.

This wasn't/isnt't meant to be a bug report or discussion about some issue in your code. In fact your code has always served me very well and it works great, thanks for that!

In my comments I was referring to some modifications I tried to apply to your code. It seemed to me that I would need to do this but everything works now as it is! 

I'll close this, as everything is working as expected.

**Comment by WebReflection** [comment]
@dsheyp but is this a TS only issue ? I am really confused about your initial post, sorry.

**Comment by dsheyp** [comment]
yes, TS only. I ran into a situation with babel. after moving an existing JS project to TS, classes which are extending EventTarget suddenly have started causing problems. as it turned out the reason is with babel EventTarget being an interface only. I then started looking for a solution for my project and came across your code which I had previously only used to polyfill EventTarget for older browsers. with TS in babel, out of the box it didn't work for me so i thought i might have to modify your code a bit.
